topdown/tokens_test.go

```diff
@@ -117,19 +112,41 @@ OHoCIHmNX37JOqTcTzGn2u9+c8NlnvZ0uDvsd1BmKPaUmjmm
 			t.Errorf("expected size 256 found %d", rsaPubKey.Size())
 		}
 	})
+	t.Run("Time", func(t *testing.T) {
+		now := time.Now()
+		ts := strconv.Itoa(int(now.UnixNano()))
+		wallclock := ast.NumberTerm(json.Number(ts))
+
+		t.Run("if provided, is parsed properly", func(t *testing.T) {
+			c := ast.NewObject()
+			c.Insert(ast.StringTerm("time"), wallclock)
+			constraints, err := parseTokenConstraints(c, ast.NumberTerm(json.Number("12134")))
+			if err != nil {
+				t.Fatalf("parseTokenConstraints: %v", err)
+			}
+			diff := now.UnixNano() - constraints.time
+			if diff > 1000 {
+				t.Errorf("expected time constraint to be within 1000ns of the one passed in, got %dns", diff)
+			}
+		})
+
+		t.Run("unset, defaults to wallclock", func(t *testing.T) {
+			c := ast.NewObject() // 'time' constraint is unset
+			constraints, err := parseTokenConstraints(c, wallclock)
+			if err != nil {
+				t.Fatalf("parseTokenConstraints: %v", err)
+			}
+			diff := now.UnixNano() - constraints.time
+			if diff > 1000 {
+				t.Errorf("expected time constraint to be within 1000ns of the one passed as default, got %dns", diff)
+			}
```

Comment:
Similar to the above, I thought this was testing that one time is chosen over the other. Do we need subtraction and the actual `time.Now` value?